realm/realm-library/src/objectServer/java/io/realm/internal/network/LookupUserIdResponse.java

```diff
@@ -0,0 +1,127 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal.network;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import io.realm.ErrorCode;
+import io.realm.ObjectServerError;
+import io.realm.log.RealmLog;
+import okhttp3.Response;
+
+/**
+ * Class wrapping the response from `GET /api/providers/:provider/accounts/:provider_id`
+ */
+public class LookupUserIdResponse extends AuthServerResponse {
+
+    private static final String JSON_FIELD_USER = "user";
+    private static final String JSON_FIELD_USER_ID = "id";
+    private static final String JSON_FIELD_USER_IS_ADMIN = "isAdmin";
+
+    private final String userId;
+    private final Boolean isAdmin;
+
+    /**
+     * Helper method for creating the proper lookup user response. This method will set the appropriate error
+     * depending on any HTTP response codes or I/O errors.
+     *
+     * @param response the server response.
+     * @return the user lookup response.
+     */
+    static LookupUserIdResponse from(Response response) {
+        String serverResponse;
+        try {
+            serverResponse = response.body().string();
+        } catch (IOException e) {
+            ObjectServerError error = new ObjectServerError(ErrorCode.IO_EXCEPTION, e);
+            return new LookupUserIdResponse(error);
+        }
+        if (!response.isSuccessful()) {
+            return new LookupUserIdResponse(AuthServerResponse.createError(serverResponse, response.code()));
+        } else {
+            return new LookupUserIdResponse(serverResponse);
+        }
+    }
+
+    /**
+     * Helper method for creating a failed response.
+     */
+    public static LookupUserIdResponse from(ObjectServerError objectServerError) {
+        return new LookupUserIdResponse(objectServerError);
+    }
+
+    /**
+     * Helper method for creating a failed response from an {@link Exception}.
+     */
+    public static LookupUserIdResponse from(Exception exception) {
+        return LookupUserIdResponse.from(new ObjectServerError(ErrorCode.fromException(exception), exception));
+    }
+
+    private LookupUserIdResponse(ObjectServerError error) {
+        RealmLog.debug("LookupUserIdResponse - Error: " + error);
+        setError(error);
+        this.error = error;
+        this.userId = null;
+        this.isAdmin = null;
+    }
+
+    private LookupUserIdResponse(String serverResponse) {
+        ObjectServerError error;
+        String userId;
+        Boolean isAdmin;
+        String message;
+        try {
+            JSONObject obj = new JSONObject(serverResponse);
+            JSONObject jsonUser = obj.getJSONObject(JSON_FIELD_USER);
+            if (jsonUser != null) {
+                userId = jsonUser.has(JSON_FIELD_USER_ID) ? jsonUser.getString(JSON_FIELD_USER_ID) : null;
```

Comment:
`jsonUser.optString(JSON_FIELD_USER_ID, null);` is a lot shorter 😉  Also below with `optBoolean`